actors/miner/tests/miner_actor_test_bitfield_queue.rs

```diff
@@ -0,0 +1,263 @@
+    queue.add_to_queue_values(epoch2, [2, 4].to_vec()).unwrap();
+
+    let (next, modified) = queue.pop_until(epoch1 - 1).unwrap();
+
+    // no values are returned
+    let count = next.len();
+    assert_eq!(0, count);
+    // modified is false
+    assert!(!modified);
+
+    let mut bq_expectation = BQExpectation::default();
+    // queue remains the same
+    bq_expectation.add(epoch1, [1, 3].to_vec()).add(epoch2, [2, 4].to_vec()).equals(&mut queue);
+}
+
+#[test]
+fn pop_until_removes_and_returns_entries_before_and_including_target_epoch() {
+    let h = ActorHarness::new(0);
+    let rt = h.new_runtime();
+    let mut queue = empty_bitfield_queue(&rt, TEST_AMT_BITWIDTH);
+
+    let epoch1 = ChainEpoch::from(42);
+    let epoch2 = ChainEpoch::from(93);
+    let epoch3 = ChainEpoch::from(94);
+    let epoch4 = ChainEpoch::from(204);
+
+    queue.add_to_queue_values(epoch1, [1, 3].to_vec()).unwrap();
+    queue.add_to_queue_values(epoch2, [5].to_vec()).unwrap();
+    queue.add_to_queue_values(epoch3, [6, 7, 8].to_vec()).unwrap();
+    queue.add_to_queue_values(epoch4, [2, 4].to_vec()).unwrap();
+
+    let (next, modified) = queue.pop_until(epoch2).unwrap();
+    // modified should be true to indicate queue has changed
+    assert!(modified);
+
+    // values from first two epochs are returned
+    assert_bitfield_equals(&next, &[1, 3, 5]);
+
+    let mut bq_expectation = BQExpectation::default();
+    // queue only contains remaining values
+    bq_expectation.add(epoch3, [6, 7, 8].to_vec()).add(epoch4, [2, 4].to_vec()).equals(&mut queue);
+
+    // subsequent call to epoch less than next does nothing
+    let (next, modified) = queue.pop_until(epoch3 - 1).unwrap();
+    assert!(!modified);
+
+    // no values are returned
+    assert_bitfield_equals(&next, &[]);
+
+    let mut bq_expectation = BQExpectation::default();
+    // queue only contains remaining values
+    bq_expectation.add(epoch3, [6, 7, 8].to_vec()).add(epoch4, [2, 4].to_vec()).equals(&mut queue);
+
+    // popping the rest of the queue gets the rest of the values
+    let (next, modified) = queue.pop_until(epoch4).unwrap();
+    assert!(modified);
+
+    // rest of values are returned
+    assert_bitfield_equals(&next, &[2, 4, 6, 7, 8]);
+
+    let mut bq_expectation = BQExpectation::default();
+    // queue is now empty
+    bq_expectation.equals(&mut queue);
+}
+
+#[test]
+fn cuts_elements() {
+    let h = ActorHarness::new(0);
+    let rt = h.new_runtime();
+    let mut queue = empty_bitfield_queue(&rt, TEST_AMT_BITWIDTH);
+
+    let epoch1 = ChainEpoch::from(42);
+    let epoch2 = ChainEpoch::from(93);
+
+    queue.add_to_queue_values(epoch1, [1, 2, 3, 4, 99].to_vec()).unwrap();
+    queue.add_to_queue_values(epoch2, [5, 6].to_vec()).unwrap();
+
+    let ranges: Vec<Range<u64>> = vec![Range { start: 2, end: 3 }, Range { start: 4, end: 7 }];
+    let to_cut = BitField::from_ranges(Ranges::new(ranges.iter().cloned()));
```

Comment:
You could write something like `MaybeBitField::from_iter(&[2,4,5,6]).unwrap()`

Reply:
I like it; that is much more concise. Modified the code to use this method.